### DISCUSSION TITLE
How do we fix the ignoreVary warnings?

### DISCUSSION THREAD

**Discussion by tony19** [serwist/serwist#65] [Q&A]
I'm using Serwist with Next 14's app router and Zenstack, and I see many `ignoreVary` warnings in the browser console:

> serwist The response for /api/model/spaceUser/findMany?q=%7B%22where%22%3A%7B%22spaceId%22%3A%22cls5ageek00014jxcjqeb93rx%22%7D%2C%22include%22%3A%7B%22user%22%3Atrue%7D%2C%22orderBy%22%3A%7B%22role%22%3A%22desc%22%7D%7D has a 'Vary: RSC, Next-Router-State-Tree, Next-Router-Prefetch, Next-Url' header. Consider setting the {ignoreVary: true} option on your strategy to ensure cache matching and deletion works as expected.

The DevTools cache storage shows that every request except `static-font-assets` has a `Vary` header, which causes all the warnings.

I tried to add the `ignoreVary: true` option to `matchOptions`, which failed to resolve the warnings:

```typescript
// src/app/sw.ts
import { defaultCache } from '@serwist/next/browser'
import type { RuntimeCaching } from '@serwist/build'
import { installSerwist } from '@serwist/sw'

// XXX: Probably not wise to enable `ignoreVary` on *every* entry,
// but we're doing this to test whether the option actually stops
// the annoying warnings in the browser console.
const configIgnoreVaryOption = (cache: RuntimeCaching[]) => {
  const newCache = cache.map(entry => {
    if (typeof entry === 'string') {
      return entry
    }

    return {
      ...entry,
      options: {
        ...entry.options,
        matchOptions: {
          ...entry.options?.matchOptions,
          ignoreVary: true,
        },
      },
    } as RuntimeCaching
  })
  return newCache
}

installSerwist({
  runtimeCaching: [
    ...configIgnoreVaryOption(defaultCache),
  ],
})
```

How do we properly address the `ignoreVary` warnings?

**Comment by DuCanhGH** [comment]
@tony19  sorry for having kept you waiting for 2 weeks. I've been fairly focused on [v9](https://github.com/serwist/serwist/discussions/66), so it's been quite hard to keep track of things.

Regarding this issue, this is working as previously intended. This warning was meant to be logged whenever this header showed up. Still, we'd like to change that in 9.0.0.

This has been changed in 9.0.0-preview.9! If you'd like to disable this log right now, you can migrate to Serwist preview (details are mentioned in the linked discussion). Note that in v9, to set `ignoreVary` to `true` like in the example code you attached, you'd need to set `entry.strategy.matchOptions.ignoreVary` rather than `entry.options.matchOptions`.
